docs/proposals/config-management-plugin-v2.md

```diff
@@ -0,0 +1,304 @@
+
+The cmp-server will expose the following APIs to the repo-server,
+
+- GenerateManifests(path) - returns YAML output using plugin tooling
+- IsSupported(path) - returns whether or not the given path is supported by the plugin
+
+At startup, cmp-server looks at the /plugin.yaml ConfigManagementPlugin specification file to understand how to perform the requests.
+
+#### Registration & Communication
+The repo-server needs to understand what all plugins are available to render manifests. To do this, the cmp-server 
+sidecars will register themselves as available plugins to the argocd-repo-server by populating named socket files in the 
+shared volume between repo-server and cmp-server. e.g.:
+
+```bash
+/var/run/argocd/plugins/
+                        cdk8s.sock
+                        jkcfg.sock
+                        pulumi.sock
+```
+
+The name of the socket file will indicate the plugin name. To discover the available plugins, the repo-server will list 
+the shared plugins directory to discover the available plugin cmp-servers running as sidecars.
+
+To communicate with a plugin, the repo-server will simply need to connect to the socket and make gRPC calls against the 
+cmp-server listening on the other side. 
+
+#### Discovery (Auto-selection of Tool)
+
+- The plugin discovery will run in the main repo-server container.
+- Argo CD repo-server lists the shared plugins directory and runs `discover` command from the specification file, 
+whichever plugin provides a positive response first will be selected.
+- In case of multiple plugins, repo-server will decide based on the plugin priority list set by argocd-server. 
+This priority list can be set in the Argo CD config map. 
+
+#### Versioning
+There will be one image per version. Hence, for two different versions users will have to configure two different sidecars.
+
+### Security Considerations
+
+The use of the plugin as sidecars separate from the repo-server is already a security improvement over the current v1.8 
+config management plugin mechanism, since the plugin tooling will no longer have access to the files of the argocd-repo-server image. 
+However additional improvements can be made to increase security.
+
+Currently, nothing prevents a command from traversing outside of the git repository, and reading/writing files outside 
+the repository path, including other git repositories which were cloned next to the working directory of the generate command. 
+Although the generate commands which are executed are determined by an operator, they are still affected by end user input. 
+For example, an  end-user might try to exploit the reposerver by specifying `HEAD && rm -rf /` as a git targetRevision, 
+which they might expect to execute as:
+
+`git clone HEAD && rm -rf /`
+
+(Note that this is just used as an example and the repo-server is not actually exploitable in this way)
```

Comment:
Since you've mentioned the potential exploit, could you mention why this is not exploitable in this way :-)